### PR TITLE
[TE] Self-serve Logic Enhancement for Import Metrics

### PR DESCRIPTION
**Pull request by newsummit**
I've refactored and ...

- added some validation for non-existent inGraph dashboard names ... still waiting for the endpoint to be created, but placeholder is there.
- made sure that the "list metrics for this new dataset name" is called AFTER the "trigger onboard" call. Otherwise, it will return empty if called before the metrics actually onboard.

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/self-serve/import-metric/controller.js [hunk 1]
```diff
@@ -78,10 +78,29 @@ export default Ember.Controller.extend({
...
-   * @method fetchAnomalyByName
-   * @param {String} functionName - name of alert or function
+   * Validates whether the entered dashboard name exists in inGraphs
+   * @method validateDashboardName
+   * @param {Boolean} isRrdImport - indicates whether we are importing custom dashboard name
+   * @param {String} dashboardName - name of dashboard to import
+   * @return {Promise}
+   */
+  validateDashboardName(isRrdImport, dashboardName) {
+    // TODO: add correct endpoint
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/self-serve/import-metric/controller.js]: is this still a todo?

File: thirdeye/thirdeye-frontend/app/pods/self-serve/import-metric/controller.js [hunk 2]
```diff
@@ -78,10 +78,29 @@ export default Ember.Controller.extend({
...
+    const url = `/onboard/validate?dashboard=${dashboardName}`
+    // Only make the call if we are importing an existing inGraphs dashboard (isRrdImport = false)
+    if (isRrdImport) {
+      return Promise.resolve(true);
+    } else {
+      // TODO: enable when endpoint ready
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/self-serve/import-metric/controller.js]: same

File: thirdeye/thirdeye-frontend/app/pods/self-serve/import-metric/controller.js [hunk 3]
```diff
@@ -158,31 +212,27 @@ export default Ember.Controller.extend({
...
-          // Trigger onboard for imported metrics
-          this.triggerInstantOnboard().then(onboardRes => {
-            this.setProperties({
-              isImportSuccess: true,
-              isMetricOnboarded: true,
-              importedMetrics: metricsList.Records
-            });
+      // Check whether provided dashboard name exists before sending onboard requests.
+      this.validateDashboardName(isRrdImport, datasetName)
+      .then((isValid) => {
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/self-serve/import-metric/controller.js]: nit: please indent `.then`s

File: thirdeye/thirdeye-frontend/app/pods/self-serve/import-metric/controller.js (hunk 1 above)
Comment (newsummit) [inline thirdeye/thirdeye-frontend/app/pods/self-serve/import-metric/controller.js]: Yes - that endpoint is fake for now.

**Pull request merged**